datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/DurationDeserializer.java

```diff
@@ -129,9 +164,13 @@ protected Duration _fromString(JsonParser parser, DeserializationContext ctxt,
         }
 
         try {
+            if (_durationUnitParser != null) {
+                return _durationUnitParser.parse(NumberInput.parseLong(value));
+            }
+
             return Duration.parse(value);
-        } catch (DateTimeException e) {
-            return _handleDateTimeException(ctxt, e, value);
+        } catch (NumberFormatException | DateTimeException e) {
+            return _handleWeirdStringValue(ctxt, e, value);
```

Comment:
@cowtowncoder  Ok to use same exception handler method, for a number? Or should be _handleWeirdNumberValue(...)?

Reply:
I think is ok to use same handler because at the end the original value that is on processing is a `String`, so it matches the handler purpose.

Reply:
since `String` is not handled anymore I reverted these changes.